net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+    ln -s /private/var/${calendarserverUser}/Library/CalendarServer/Data \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/Data
+
+    # source
+    move ${worksrcpath} \
+        ${destroot}${prefix}/src/${ccsname}
+
+    # calenderserver configuration
+    xinstall -m 0600 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/calendarserver.plist \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/Config/calendarserver.plist.macports
+
+    # nginx reverse proxy
+    xinstall -m 0640 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/nginx.conf \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx.conf.macports
+    xinstall -m 0644 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/default.html \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx_root
+
+    # ccs-calendarserver's postgres code points to `PSQL = "../postgresql/_root/bin/psql"`
+    ln -s ${prefix} \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/postgresql/_root
+}
+
+# Network configuration
+# hard-coded examples
+set host                host
+set domain              domain
+set tld                 tld
+set fullhost            ${host}.${domain}.${tld}
+set domaintld           ${domain}.${tld}
+set tls_certificate_name \
+                        ${fullhost}
+
+startupitem.create     yes
+
+startupitems \
+    name               calendarserver \
+    executable \
+        ${calendarserverdir}/Library/CalendarServer/${ccsname}/bin/caldavd
+
+# use defaults, edit launch daemon plists in post-activate below
+# executable ${prefix}/sbin/nginx -g "daemon off;" -p ${calendarserverdir}/Library/CalendarServer/logs -c ${calendarserverdir}/Library/CalendarServer/etc/nginx.conf
+# pidfile auto ${calendarserverdir}/Library/CalendarServer/logs/nginx.pid
+startupitems-append \
+    name               calendarserver_proxy \
+    executable         ${prefix}/sbin/nginx \
+    pidfile            auto
+
+post-activate {
+    # personal data outside of ${prefix}
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer] } {
+        xinstall -m 0755 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer
+    }
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer/Data] } {
+        xinstall -m 0700 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer/Data
+    }
+
+    # delete previous install (see destroot notes above)
+    foreach f_or_d {
+        bin
+        environment.sh
+        lib
+        roots
+        virtualenv
+        } {
+        if { [file exists \
+                  ${calendarserverdir}/Library/CalendarServer/${ccsname}/${f_or_d}]
+            } {
+            delete \
+                ${calendarserverdir}/Library/CalendarServer/${ccsname}/${f_or_d}
```

Comment:
This is supposed to happen at deactivation.

Reply:
ccs-calendar has no `DESTDIR` capability and might be said to have anti-`DESTDIR` behavior.

Remember, this is the code Apple used to package up Calendar and Contacts Server for Server.app, so they didn’t care so much about how to deal with package managers.

The script `./bin/package` hardcodes the directory it’s built in into all compiled Python code and object files. Doing this in `destroot` completely breaks the build, either because ccs-calendarserver sticks expanded `${destroot}` into everything, or there are insufficient macports permissions to build and install things where they need to go in stages before `activate`.

Yes, it’s hack. But it works, it’s isolated to `./bin/package` results, it’s not too bad, and it works.

See https://github.com/apple/ccs-calendarserver/issues/540